src/Umbraco.Core/Configuration/GlobalSettings.cs

```diff
@@ -378,5 +378,43 @@ public bool UseHttps
                 }
             }
         }
+
+        private string _umbracoMediaPath = null;
+        public string UmbracoMediaPath => GetterWithDefaultValue(Constants.AppSettings.UmbracoMediaPath, "~/umbraco", ref _umbracoMediaPath);
+
+        private string _umbracoScriptsPath = null;
+        public string UmbracoScriptsPath => GetterWithDefaultValue(Constants.AppSettings.UmbracoScriptsPath, "~/umbraco", ref _umbracoScriptsPath);
+
+        private string _umbracoCssPath = null;
+        public string UmbracoCssPath => GetterWithDefaultValue(Constants.AppSettings.UmbracoCssPath, "~/umbraco", ref _umbracoCssPath);
+
+        private string _umbracoPath = null;
+        public string UmbracoPath => GetterWithDefaultValue(Constants.AppSettings.UmbracoPath, "~/umbraco", ref _umbracoPath);
+
+        private T GetterWithDefaultValue<T>(string appSettingKey, T defaultValue, ref T backingField)
+        {
+            if (backingField != null) return backingField;
+
+            if (ConfigurationManager.AppSettings.ContainsKey(appSettingKey))
+            {
+                try
+                {
+                    var value = ConfigurationManager.AppSettings[appSettingKey];
+
+                    backingField = (T)Convert.ChangeType(value, typeof(T));
```

Comment:
Might be better to use our `TryConvertTo` ext method which is a lot smarter than the ChangeType conversion, plus then you don't need to try/catch